### DISCUSSION TITLE
Find a name of application to pass bind

### DISCUSSION THREAD

**Discussion by omnigenous** [hyprwm/Hyprland#4288] [Q&A]
Already checked [docs](https://wiki.hyprland.org/Configuring/Binds/#global-keybinds)

But I can't figure out how to find a name for applications to pass binds

For TeamSpeak3 docs use `"^(TeamSpeak 3)$"` which seems to be a title of a window at all the time which never change.

OBS have a lot of different information on it window title so I guess that is why `"^(com\.obsproject\.Studio)$"` used but I can't figure out where `"^(com\.obsproject\.Studio)$"` came from. Where do I find _it_ (and also what do you call _it_?) for other applications?

I'm trying to pass a key to [YouTube Music](https://github.com/th-ch/youtube-music)
```bash
bind = , <KEY>, pass, ^(YouTube Music)$
```

**Comment by omnigenous** [comment]
Okay, I got it working...kinda

I enabled "Shortcuts" Plugin in YouTube Music app:
![2023-12-29_03-43-28](https://github.com/hyprwm/Hyprland/assets/154441415/f49f0401-b02d-4713-89c5-c67a85ed2363)

Added bind to hyprland config:
```bash
bind = , XF86AudioPlay, pass, ^(.*YouTube Music)$
```

Problem is bind only work when I either have YouTube Music window active or last active window was in same workspace as YouTube Music app. If I switch workspace and select any window in that workspace bind won't work.

**Comment by omnigenous** [comment]
Seems like it some sort of x11/wayland/xwayland issue. YouTube Music is running on x11 so as long as last hovered window is also x11 and not wayland I can use my global bind.

**Comment by ErrorNoInternet** [comment]
`hyprctl clients`

**Comment by festeh** [comment]
I would also like to know how to enable global shortcuts from everywhere for apps with `GDK_BACKEND=x11`. For now I use `hyprctl dispatch` commands to move/focus a window, but this is kinda hacky.

**Comment by omnigenous** [comment]
@festeh I'm still new to hyprland could you share more details on how to get it to work?

**Comment by festeh** [comment]
I do it like this, but that is probably not the best option

In my case I just need to open/hide an app globally. Suppose it is named `Foo`. I then have two scripts, like `openfoo` and `hidefoo` in `/usr/bin`

`openfoo`
```
#!/bin/bash
W_ID=$(hyprctl activeworkspace -j | jq .id)
hyprctl dispatch movetoworkspace "$W_ID,Foo"
```

`hidefoo`
```
#!/bin/bash
hyprctl dispatch movetoworkspacesilent "9,Foo"
```

This setup assumes that `Foo` is always opened in workspace 9, which is never used. When I need the actual app, I run `openfoo` ti move it to currently opened workspace, when I am done it is moved back with `hidefoo`, both can be assigned to hotkeys.

## Your case
Your scenario is probably more complicated, so it might not work very well, but if you want to try, here are some steps.
1) Move you window to active workspace, like in my case
2) Focus it, check `focuswindow` [here](https://wiki.hyprland.org/Configuring/Dispatchers/)
3) Either press you shortcut, or if you want more automation check [ydotool](https://github.com/ReimuNotMoe/ydotool)
4) Move window back to some unused workspace

A possible issue here is some flickering because of moving back and forth windows, so it would be nice to see better solutions.
